Sources/AuthFoundation/User Management/Credential.swift

```diff
@@ -111,15 +135,15 @@ public final class Credential: Equatable, OAuth2ClientDelegate {
     /// The ID the token is identified by within storage.
     ///
     /// This value is automatically generated when the token is stored, and is a way to uniquely identify a token within storage. This value corresponds to the IDs found in the ``allIDs`` static property.
-    public lazy var id: String = { token.id }()
+    public var id: String { token.id }
     
     /// The metadata associated with this credential.
     ///
     /// This property can be used to associate application-specific information with a ``Token``. This can be used to identify which token should be associated with certain parts of your application.
     ///
     /// > Important: Errors thrown from the setter are silently ignored. If you would like to handle errors when changing metadata, see the ``setTags(_:)`` function.
     public var tags: [String: String] {
-        get { _metadata.tags }
+        get { metadata.tags }
```

Comment:
withAsyncGroup?

Reply:
The `metadata` property has custom getter/setter which uses the credential's local `lock` to synchronize access to the `nonisolated(unsafe) _metadata` property. From there, the Metadata type is `Sendable`, and its `tags` property is a `let` value, so it's safe to access across isolation contexts.